projects/max14906/main.c

```diff
@@ -0,0 +1,332 @@
+#include <stdio.h>
+
+#include "parameters.h"
+#include "no_os_uart.h"
+#include "no_os_print_log.h"
+#include "no_os_delay.h"
+#include "maxim_spi.h"
+#include "maxim_gpio.h"
+#include "maxim_uart.h"
+#include "maxim_uart_stdio.h"
+#include "max14906.h"
+#include <string.h>
+
+#ifdef IIO_SUPPORT
+#include "iio_max14906.h"
+#include "iio_app.h"
+#endif
+
+/** Define UART and SPI macro's corresponding to the uController used. */
+#define UART_IRQ_ID		UART1_IRQn
+#define UART_DEVICE_ID		1
+#define UART_BAUDRATE		57600
+#define UART_EXTRA		&max14906_uart_extra
+#define UART_OPS		&max_uart_ops
+
+#define SPI_DEVICE_ID		1
+#define SPI_OPS			&max_spi_ops
+#define SPI_EXTRA		&max14906_spi_extra
+#define SPI_CS			0
+#define SPI_BAUDRATE		100000
+
+/** Defining specific mask for this example. */
+#define MAX14906_SLED_CH(x)	NO_OS_BIT(x + MAX14906_CHANNELS)
+
+/** Buffer macros and variable for iio. */
+#define DATA_BUFFER_SIZE	400
+
+uint16_t iio_data_buffer[DATA_BUFFER_SIZE*4*sizeof(int)];
+
+/** MAX specific uart and spi. */
+struct max_uart_init_param max14906_uart_extra = {
+	.flow = UART_FLOW_DIS,
+	.vssel = MXC_GPIO_VSSEL_VDDIO,
+};
+
+struct max_spi_init_param max14906_spi_extra = {
+	.num_slaves = 1,
+	.polarity = SPI_SS_POL_LOW,
+	.vssel = MXC_GPIO_VSSEL_VDDIOH,
+};
+
+struct max_gpio_init_param gpio_extra_ip = {
+	.vssel = MXC_GPIO_VSSEL_VDDIO
+};
+
+static const char *climit_name[] = {
+	"600mA", "130mA", "300mA", "1.2A",
+};
+
+/** NO-OS uart and spi. */
+struct no_os_uart_init_param max14906_uart_ip = {
+	.device_id = UART_DEVICE_ID,
+	.baud_rate = UART_BAUDRATE,
+	.size = NO_OS_UART_CS_8,
+	.platform_ops = UART_OPS,
+	.parity = NO_OS_UART_PAR_NO,
+	.stop = NO_OS_UART_STOP_1_BIT,
+	.extra = UART_EXTRA,
+};
+
+struct no_os_spi_init_param max14906_spi_ip = {
+	.device_id = SPI_DEVICE_ID,
+	.extra = SPI_EXTRA,
+	.max_speed_hz = SPI_BAUDRATE,
+	.platform_ops = SPI_OPS,
+	.chip_select = SPI_CS,
+};
+
+struct no_os_gpio_init_param max14906_fault_gpio_param = {
+	.port = 0,
+	.pull = NO_OS_PULL_NONE,
+	.number = 5,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra_ip,
+};
+
+void gpio_callback_fn(void *ctx)
+{
+	struct max149x6_desc *desc = ctx;
+	int ret;
+	uint32_t val;
+
+	ret = max149x6_reg_read(desc, MAX14906_INT_REG, &val);
+	if (val)
+		pr_info("Fault detected!\n");
+}
+
+int main(void)
+{
+	int ret;
+
+	/** Basic example (doesn't include the iio driver.) */
+#ifndef IIO_SUPPORT
```

Comment:
Both examples were requested in the same main.c, therefore I used the old project structure.

Reply:
In this case, the project will have to live on a separate branch. We can't merge it to master in this form, since it will make further maintenance more error-prone.

Reply:
I will use the new project structure after all.